src/jquery.interactive-world-map.js

```diff
@@ -1,142 +1,180 @@
-            backgroundColor : 'white',
-            regionStyle: {
-
-                initial: {
-                    fill            :   'url(#dotsGeneral)'
-                },
-
-                hover: {
-                    "fill-opacity":  1
-                }
-            },
-            series: {
-                regions: [{
-                    values: countriesFill,
-                    attribute: 'fill',
-                    hover: {
-                        "fill-opacity":  0.8,
-                        cursor:         'pointer'
-                    }
-                }]
-            },
-            onRegionOver: function (e, code) {
-                if(dataCountries[code]){
-                    data = new Array;
-                    data[code] = 'url(#dotsHover)';
-                    var mapObject = $(this).parent().vectorMap('get', 'mapObject');
-                    mapObject.series.regions[0].setValues(data);
-                }
-            },
-            onRegionOut: function (e, code) {
-                if(dataCountries[code]) {
-                    data = new Array;
-                    data[code] = 'url(#dotsHighlighted)';
-                    var mapObject = $(this).parent().vectorMap('get', 'mapObject');
-                    mapObject.series.regions[0].setValues(data);
-                }
-            },
-
-            onRegionClick: function(e, code){
-                if(!dataCountries[code]) {
-                    return;
-                }
-                var $this = $(this);
-                $element   = $('g', $this).find("[data-code='" + code  + "']");
-                $offset    = $element.offset();
-
-                $(".main-modal, #modal-background").toggleClass("active");
-                // Calculate the distance between the country and the top for show with the correctly space the popup
-                distanceXtop = $offset.top  - marginCountryPopupForTop;
-                distanceLeft = $offset.left + marginCountryPopupForLeft;
-
-                var first           = dataCountries[code];
-                countryName         = first['name'];
-                description         = first['description'];
-                mainImage           = first['main_image'];
-
-                $(".main-modal" ).css("top", distanceXtop + 'px').css("left", distanceLeft + 'px');
-                $(".main-modal .description").text(description);
-                $(".main-modal .title").text(countryName);
-                $(".main-image").attr('src', mainImage);
+    function setPatternAttributes(x, y, cx, cy, r) {
+        _x  = x;
+        _y  = y;
+        _cx = cx;
+        _cy = cy;
+        _r  = r;
+    }
 
-            }
+    function createPatternSvg(id, color) {
+        return '<pattern id="' + id + '" width="5" height="5"     patternUnits="userSpaceOnUse"><circle x="' + _x + '" y="' + _y + '" cx="' + _cx + '" cy="' + _cy + '" r="' + _r + '" style="stroke:none; fill:' + color + '"></circle></pattern>';
+    }
```

Comment:
basically :+1: , but `width` and `height` should also be configurable. Find a **good**  hash structure that allows you to fetch the information you need, easily, without having to maintain so many variables.